keel-bakery-plugin/src/main/kotlin/com/netflix/spinnaker/keel/bakery/resource/ImageHandler.kt

```diff
@@ -76,10 +72,34 @@ class ImageHandler(
       }
     }
 
-  private fun DeliveryArtifact.findLatestVersion(statuses: List<ArtifactStatus>): String =
-    artifactRepository
-      .versions(this, statuses)
-      .firstOrNull() ?: throw NoKnownArtifactVersions(this)
+  /**
+   * First checks our repo, and if a version isn't found checks igor.
+   */
+  private fun DeliveryArtifact.findLatestVersion(statuses: List<ArtifactStatus>): String {
+    try {
+      val knowVersion = artifactRepository
+        .versions(this, statuses)
+        .firstOrNull()
+      if (knowVersion != null) {
+        return knowVersion
+      }
+    } catch (e: NoSuchArtifactException) {
+      if (!artifactRepository.isRegistered(name, type)) {
+        // we clearly care about this artifact, let's register it.
+        publisher.publishEvent(ArtifactRegisteredEvent(this, statuses))
+      }
+    }
+
+    // even though the artifact isn't registered we should grab the latest version to use
+    return runBlocking {
```

Comment:
So in this case we just return the latest version?

Reply:
yep, same as what we would get from the artifact repository if it had been listening to the versions.